api/v1alpha1/httproute_types.go

```diff
@@ -117,13 +119,20 @@ type HTTPRouteMatch struct {
 
 	// HeaderType defines the semantics of the `Header` matcher.
 	//
+	// Support: core (Exact)
+	// Support: extended (RegularExpression)
+	// Support: custom (ImplementationSpecific)
+	//
+	// Default: "Exact"
+	//
 	// +optional
 	HeaderType *string `json:"headerType" protobuf:"bytes,3,opt,name=headerType"`
```

Comment:
I'd suggest that we call this type `HeaderMatchType`, since `HeaderType` can be read as the type of the header, rather than the type of the match.